aspnet/fundamentals/owin.rst

```diff
@@ -1,8 +1,329 @@
-.. include:: /../common/stub-topic.txt
+OWIN
+====
 
-|stub-icon| OWIN
-================
+By `Steve Smith`_
 
-.. include:: /../common/stub-notice.txt
+ASP.NET 5 supports OWIN, the Open Web Interface for .NET, which allows web applications to be decoupled from web servers. In addition, OWIN defines a standard way for middleware to be used in a pipeline to handle individual requests and associated responses. ASP.NET 5 applications and middleware can interoperate with OWIN-based applications, servers, and middleware.
 
-.. _issue: https://github.com/aspnet/Docs/issues/71
\ No newline at end of file
+In this article:
+	- `Running OWIN middleware in the ASP.NET pipeline`_
+	- `Using ASP.NET Hosting on an OWIN-based server`_
+	- `Run ASP.NET 5 on an OWIN-based server and use its WebSockets support`_
+	- `OWIN keys`_
+
+`Browse or download samples on GitHub <https://github.com/aspnet/Docs/tree/master/docs/fundamentals/owin/sample>`_.
+
+Running OWIN middleware in the ASP.NET pipeline
+-----------------------------------------------
+
+ASP.NET 5's OWIN support is deployed as part of the `Microsoft.AspNet.Owin <https://www.nuget.org/packages/Microsoft.AspNet.Owin/1.0.0-beta5>`_ package, and the source is in the `HttpAbstractions repository <https://github.com/aspnet/HttpAbstractions/tree/1.0.0-beta5/src/Microsoft.AspNet.Owin>`_. You can import OWIN support into your project by adding this package as a dependency in your ``project.json`` file, as shown here:
+
+.. literalinclude:: owin/sample/src/OwinSample/project.json
+	:language: javascript
+	:lines: 5-10
+	:emphasize-lines: 3
+
+OWIN middleware conform to the `OWIN specification <http://owin.org/spec/spec/owin-1.0.0.html>`_, which defines a ``Properties IDictionary<string, object>`` interface that must be used, and also requires certain keys be set (such as ``owin.ResponseBody``). We can construct a very simple example of middleware that follows the OWIN specification to display "Hello World", as shown here:
```

Comment:
That's how the spec refers to it. Though I suppose I could move 'Properties' out of the code highlight. Here's the extract from the spec:
When the host process starts there are a number of steps it goes through to set up the application. 

1.    The host creates a Properties IDictionary<string, object> and populates any startup data or capabilities provided by the host.

2.    The host selects which server will be used and provides it with the Properties collection so it can similarly announce any capabilities.